source/features/notifications-filter.tsx

```diff
@@ -0,0 +1,146 @@
+			if (category === 'Read') {
+				if (categoryFilters.length === 1 && !notification.classList.contains(categoryFilters[0])) {
+					isNotificationHidden = true;
+					break;
+				}
+
+				continue;
+			}
+
+			if (categoryFilters.length > 0 && !select.exists(categoryFilters, notification)) {
+				isNotificationHidden = true;
+				break;
+			}
+		}
+
+		notification.hidden = isNotificationHidden;
+		const selectCheckbox = select('.js-notification-bulk-action-check-item', notification)!;
+		if (isNotificationHidden) {
+			// Prevent hidden notifications from being selected by clicking "Select all" or pressing <a>
+			selectCheckbox.removeAttribute('data-check-all-item');
+		} else {
+			/* eslint-disable-next-line unicorn/prefer-dom-node-dataset -- For consistency with the line above */
+			selectCheckbox.setAttribute('data-check-all-item', '');
+		}
+	}
+
+	// Hide empty notifications groups
+	for (const group of select.all('.js-notifications-group')) {
+		group.hidden = !select.exists('.notifications-list-item:not([hidden])', group);
+	}
+}
+
+const icons: Record<Filter, JSX.Element> = {
+	PR: <GitPullRequestIcon className="color-text-secondary"/>,
+	Issue: <IssueOpenedIcon className="color-text-secondary"/>,
+	Open: <CheckCircleIcon className="color-text-success"/>,
+	Closed: <XCircleIcon className="color-text-danger"/>,
+	Draft: <GitPullRequestDraftIcon className="color-text-tertiary"/>,
+	Merged: <GitMergeIcon className="text-purple"/>,
+	Read: <DotIcon className="color-text-link"/>,
+	Unread: <DotFillIcon className="color-text-link"/>
+};
+
+function createDropdownList(category: Category, filters: Filter[]): JSX.Element {
+	return (
+		<div className="SelectMenu-list">
+			<header className="SelectMenu-header">
+				<span className="SelectMenu-title">{category}</span>
+			</header>
+			{filters.map(filter => (
+				<label
+					className="SelectMenu-item text-normal"
+					role="menuitemcheckbox"
+					aria-checked="false"
+					tabIndex={0}
+					data-filter={filter}
+					data-category={category}
+				>
+					<CheckIcon className="octicon octicon-check SelectMenu-icon SelectMenu-icon--check mr-2" aria-hidden="true"/>
+					<div className="SelectMenu-item-text">
+						<input hidden type="checkbox"/>
+						{icons[filter]}
+						<span className="ml-2">{filter}</span>
+					</div>
+				</label>
+			))}
+		</div>
+	);
+}
+
+function createDropdown(): JSX.Element {
+	return (
+		<details className="details-reset details-overlay position-relative d-none d-md-block">
+			<summary
+				className="btn ml-3"
+				data-hotkey="f"
+				aria-haspopup="menu"
+				role="button"
+			>
+				Filter by <span className="dropdown-caret ml-1"/>
+			</summary>
+			<details-menu
+				className="SelectMenu left-0"
+				aria-label="Filter by"
+				role="menu"
+				on-details-menu-selected={handleSelection}
+			>
+				<div className="SelectMenu-modal">
+					{createDropdownList('Type', ['PR', 'Issue'])}
```

Comment:
First of all, this feature is awesome 🔥

However, these item texts should be more inline with GitHub (`Pull requests` and `Issues`) I think? You can compare it to the existing discussions (issues/PR) filter.